akka-actor/src/main/scala/akka/event/LoggerMailbox.scala

```diff
@@ -43,6 +43,5 @@ private[akka] class LoggerMailbox(owner: ActorRef, system: ActorSystem)
         envelope = dequeue
       }
     }
-    super.cleanUp(owner, deadLetters)
```

Comment:
what kind of noise would that be? have you seen it? after the above loop the mailbox is empty, i.e. `hasMessages` is `false`

note that `cleanUp` is called after switching out the mailbox, i.e. there will be no new messages added